### ISSUE TITLE
`npm repo` runs prepublish

### ISSUE THREAD

**Issue by aredridel** [npm/npm#9308]
This surprised me very much!

```
:; npm repo

> newww@2.1.0 prepublish /Users/aredridel/Projects/npm/newww
> gulp build

module.js:338
    throw err;
    ^
Error: Cannot find module 'gulp-rev-all'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/aredridel/Projects/npm/newww/gulpfile.js:14:14)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)

npm ERR! addLocal Could not install /Users/aredridel/Projects/npm/newww
npm ERR! Darwin 15.0.0
npm ERR! argv "node" "/usr/local/bin/npm" "repo"
npm ERR! node v0.12.7
npm ERR! npm  v3.3.0
npm ERR! code ELIFECYCLE
npm ERR! newww@2.1.0 prepublish: `gulp build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the newww@2.1.0 prepublish script 'gulp build'.
npm ERR! This is most likely a problem with the newww package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp build
npm ERR! You can get their info via:
npm ERR!     npm owner ls newww
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/aredridel/Projects/npm/newww/npm-debug.log
```

**Comment by othiym23** [comment]
That _is_ very surprising! Did you hold onto the `npm-debug.log`? I'm sure this should be rep(r)oducible, but it might save us a little time in figuring out why the lifecycle was getting involved at all.

**Comment by aredridel** [comment]
Here you are!

```
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/local/bin/npm', 'repo', '.' ]
2 info using npm@3.3.0
3 info using node@v0.12.7
4 verbose node symlink /usr/local/bin/node
5 silly fetchPackageMetaData .
6 silly fetchOtherPackageData .
7 silly cache add args [ '.', null ]
8 verbose cache add spec .
9 silly cache add parsed spec { raw: '.',
9 silly cache add   scope: null,
9 silly cache add   name: null,
9 silly cache add   rawSpec: '.',
9 silly cache add   spec: '/Users/aredridel/Projects/npm/newww',
9 silly cache add   type: 'directory' }
10 verbose addLocalDirectory /Users/aredridel/.npm/newww/2.1.0/package.tgz not in flight; packing
11 info prepublish newww@2.1.0
12 verbose unsafe-perm in lifecycle true
13 info newww@2.1.0 Failed to exec prepublish script
14 error addLocal Could not install /Users/aredridel/Projects/npm/newww
15 silly fetchPackageMetaData Error: newww@2.1.0 prepublish: `gulp build`
15 silly fetchPackageMetaData spawn ENOENT
15 silly fetchPackageMetaData     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:17:16)
15 silly fetchPackageMetaData     at ChildProcess.emit (events.js:110:17)
15 silly fetchPackageMetaData     at maybeClose (child_process.js:1015:16)
15 silly fetchPackageMetaData     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
15 silly fetchPackageMetaData  error for . { [Error: newww@2.1.0 prepublish: `gulp build`
15 silly fetchPackageMetaData spawn ENOENT]
15 silly fetchPackageMetaData   code: 'ELIFECYCLE',
15 silly fetchPackageMetaData   errno: 'ENOENT',
15 silly fetchPackageMetaData   syscall: 'spawn',
15 silly fetchPackageMetaData   file: 'sh',
15 silly fetchPackageMetaData   pkgid: 'newww@2.1.0',
15 silly fetchPackageMetaData   stage: 'prepublish',
15 silly fetchPackageMetaData   script: 'gulp build',
15 silly fetchPackageMetaData   pkgname: 'newww' }
16 verbose stack Error: newww@2.1.0 prepublish: `gulp build`
16 verbose stack spawn ENOENT
16 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:17:16)
16 verbose stack     at ChildProcess.emit (events.js:110:17)
16 verbose stack     at maybeClose (child_process.js:1015:16)
16 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
17 verbose pkgid newww@2.1.0
18 verbose cwd /Users/aredridel/Projects/npm/newww
19 error Darwin 15.0.0
20 error argv "node" "/usr/local/bin/npm" "repo" "."
21 error node v0.12.7
22 error npm  v3.3.0
23 error file sh
24 error code ELIFECYCLE
25 error errno ENOENT
26 error syscall spawn
27 error newww@2.1.0 prepublish: `gulp build`
27 error spawn ENOENT
28 error Failed at the newww@2.1.0 prepublish script 'gulp build'.
28 error This is most likely a problem with the newww package,
28 error not with npm itself.
28 error Tell the author that this fails on your system:
28 error     gulp build
28 error You can get their info via:
28 error     npm owner ls newww
28 error There is likely additional logging output above.
29 verbose exit [ 1, true ]
```

**Comment by aredridel** [comment]
(actually started a new log, so slightly different error, but nice and short)

**Comment by othiym23** [comment]
Fixed via f4a578424b1f9d12f58210c229e7af633d952587, included in `npm@3.3.1`. Thanks for the ace bug report, @aredridel!
